File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/api/Models2Api.java
package com.sdkwork.ai.api;

import com.sdkwork.ai.http.HttpClient;
import com.sdkwork.ai.model.*;
import java.util.List;
import java.util.Map;

public class Models2Api {
    private final HttpClient client;
    
    public Models2Api(HttpClient client) {
        this.client = client;
    }

    /** Retrieve model */
    public OpenAiModel retrieveModel(String model) throws Exception {
        return (OpenAiModel) client.get(ApiPaths.aiPath("/v1/models/" + model + ""));
    }

    /** Delete model */
    public OpenAiModelDeleteResponse deleteModel(String model) throws Exception {
        return (OpenAiModelDeleteResponse) client.delete(ApiPaths.aiPath("/v1/models/" + model + ""));
    }

    /** List models */
    public OpenAiModelList listModels() throws Exception {
        return (OpenAiModelList) client.get(ApiPaths.aiPath("/v1/models"));
    }
}

File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/api/Models2Api.kt
package com.sdkwork.ai.api

import com.sdkwork.ai.*
import com.sdkwork.ai.http.HttpClient

class Models2Api(private val client: HttpClient) {

    /** Retrieve model */
    suspend fun retrieveModel(model: String): OpenAiModel? {
        return client.get(ApiPaths.aiPath("/v1/models/$model")) as? OpenAiModel
    }

    /** Delete model */
    suspend fun deleteModel(model: String): OpenAiModelDeleteResponse? {
        return client.delete(ApiPaths.aiPath("/v1/models/$model")) as? OpenAiModelDeleteResponse
    }

    /** List models */
    suspend fun listModels(): OpenAiModelList? {
        return client.get(ApiPaths.aiPath("/v1/models")) as? OpenAiModelList
    }
}
